### DISCUSSION TITLE
Missing biomass data for Netherlands (NL)

### DISCUSSION THREAD

**Discussion by lithiumavenger** [electricitymaps/electricitymaps-contrib#4102] [General]
Dear all,

I'm very new to Github and coding, but I noticed that for the Netherlands, biomass data is not visible on the map. It seems like coal + biomass data is combined and showed as coal. This results in a much higher calculated CO2 emission than if both were split, as NL uses a lot of biomass: (souces: https://energieopwek.nl and https://longreads.cbs.nl/the-netherlands-in-numbers-2020/how-much-energy-do-we-get-from-biomass/.

I think the biomass data are the sources used for ElectricityMap (https://energieopwek.nl provides live biomass electricity generation data), but there might be a mistake in that part of the code?

**Comment by VIKTORVAV99** [comment]
Hi and welcome to GitHub @lithiumavenger!

It looks like the data source for NL is a combination of https://energieopwek.nl/ and [ENTSO-E transparency data](https://transparency.entsoe.eu/generation/r2/actualGenerationPerProductionType/show?name=&defaultValue=true&viewType=TABLE&areaType=BZN&atch=false&datepicker-day-offset-select-dv-date-from_input=D&dateTime.dateTime=03.05.2022+00:00|CET|DAYTIMERANGE&dateTime.endDateTime=03.05.2022+00:00|CET|DAYTIMERANGE&area.values=CTY|10YNL----------L!BZN|10YNL----------L&productionType.values=B01&productionType.values=B02&productionType.values=B03&productionType.values=B04&productionType.values=B05&productionType.values=B06&productionType.values=B07&productionType.values=B08&productionType.values=B09&productionType.values=B10&productionType.values=B11&productionType.values=B12&productionType.values=B13&productionType.values=B14&productionType.values=B20&productionType.values=B15&productionType.values=B16&productionType.values=B17&productionType.values=B18&productionType.values=B19&dateTime.timezone=CET_CEST&dateTime.timezone_input=CET+(UTC+1)+/+CEST+(UTC+2)) and for biomass specifically ENTSO-E data is used. (I hope I got this right, the parser is a bit more complex as it uses two sources.)

Now ENTSO-E data comes from local TSOs (Transmission System Operators), I believe it is www.tennet.eu for NL, so if you believe there is a error in this data it would be best to contact ENTSO-E or Tennet (if that's the correct one) directly.

It might be possible switch to https://energieopwek.nl/ for NL biomass data though but I don't know what implications of that will be so I will leave that question for someone else.

**Reply by lithiumavenger** [reply]
Thanks for your response! In the [ENTSO-E translarency data ](https://transparency.entsoe.eu/generation/r2/actualGenerationPerProductionType/show?name=&defaultValue=true&viewType=TABLE&areaType=BZN&atch=false&datepicker-day-offset-select-dv-date-from_input=D&dateTime.dateTime=03.05.2022+00:00%7CCET%7CDAYTIMERANGE&dateTime.endDateTime=03.05.2022+00:00%7CCET%7CDAYTIMERANGE&area.values=CTY%7C10YNL----------L!BZN%7C10YNL----------L&productionType.values=B01&productionType.values=B02&productionType.values=B03&productionType.values=B04&productionType.values=B05&productionType.values=B06&productionType.values=B07&productionType.values=B08&productionType.values=B09&productionType.values=B10&productionType.values=B11&productionType.values=B12&productionType.values=B13&productionType.values=B14&productionType.values=B20&productionType.values=B15&productionType.values=B16&productionType.values=B17&productionType.values=B18&productionType.values=B19&dateTime.timezone=CET_CEST&dateTime.timezone_input=CET+(UTC+1)+/+CEST+(UTC+2)), biomass is indeed missing. I will contact Tennet about that. In the meantime, searching for a possibility to switch to [Energieopwek ](https://energieopwek.nl/) as a source for biomass data would be more accurate. To be continued!

**Comment by q--** [comment]
ElectricityMap used Energieopwek data in the past, but [it emerged they required a paid license](https://github.com/electricitymap/electricitymap-contrib/issues/3015#issuecomment-901670043)
